### DISCUSSION TITLE
mail reports can't include alerts from worker nodes

### DISCUSSION THREAD

**Discussion by w1552270** [wazuh/wazuh#14511] [Q&A] [answered]
|Wazuh version|Component|Install type|Install method|Platform|
|4.2.3|reports|Manager|Packages|Centos7|

Automated reports in ossec.conf only include alerts processed on the master node (i.e. directly connected agents or devices sending syslog to that node). Alerts from worker nodes appear in the Elasticsearch index but are not processed for reporting.

Configuring reports on worker nodes does not work nor is documented as a possibility.

My setup is a master serving the majority of the network and two segments connected to worker nodes, I can't generate reports for logins and FIM on these

**Comment by mdiego92** [comment, marked as answer]
Hello w1552270,

- Could you please confirm if you have the same email integration in your `ossec.conf `file in your worker node?
- Also, we would like to know if you're referring to the [automatic reports](https://documentation.wazuh.com/current/user-manual/manager/automatic-reports.html) or to the [email alerts](https://documentation.wazuh.com/current/user-manual/manager/manual-email-report/)?

Thank you very much.

**Reply by w1552270** [reply]
Hi, 
I am using the "reports" tag in the ossec.conf file. My master sends them daily as expected and are generally on point but do not report anything from the networks connected to the workers. 
I tested by copying that part of configuration, the servers themselves are configured the same way to send emails and I tested that it works so I just assumed that worker nodes can't send reports.

**Reply by mdiego92** [reply]
Hello w1552270,

If you copied the working configuration from your master node it should be working.

- Could you please confirm if the `maild` and `monitord` services are running within your worker nodes?

`/var/ossec/bin/wazuh-control status
`


- Also, it would be great if you could send us the output of the following command replacing the report title parameter in your worker nodes:


`cat /var/ossec/logs/ossec.log | grep -i "report title"
`

Thank you very much.

**Reply by w1552270** [reply]
so I looked into logs and found that both workers were processing the reports but maild was not running. 
The issue is the daemon is not started when the process is restarted via systemctl. Using wazuh control restart also spun up maild.

**Reply by mdiego92** [reply]
Hello w1552270,

Could you please confirm if you could start up the `maild` service?

If you execute `cat /var/ossec/logs/ossec.log | grep -i "wazuh-maild"` do you get any error message?

Thank you very much.

**Reply by w1552270** [reply]
No I don’t get any error from maild. I know it is running and that the server can reach the smtp server but I only see: 

-Aug 26, 2022 @ 00:03:03.000 wazuh-monitord WARNING Unable to send report email. 
-Aug 26, 2022 @ 00:02:55.000 wazuh-monitord WARNING Report taking too long to complete. Waiting for it to finish...

The other worker had no error and simply didn't send anything because the reports were empty (this is expected, I am triggering events to test this)

I will try to enable debugging messages next week and see exactly why it fails.

**Reply by mdiego92** [reply]
Hello w1552270,

Is your manager handling a large number of alerts in the `alerts.log/json` file? It would be good to know if you're receiving emails after a few minutes/hours or if you don't receive emails at all.

Could you please try enabling the `<json_logall>` setting in your `ossec.conf` file and check if there are any relevant log entries?

Thank you very much!
